cni/pkg/install-cni/pkg/install/install.go

```diff
@@ -168,10 +172,12 @@ func sleepCheckInstall(ctx context.Context, cfg *config.Config, cniConfigFilepat
 			log.Infof("Invalid configuration. %v", checkErr)
 			return nil
 		}
+		SetReady(isReady)
 		// Valid configuration; Wait for modifications before checking again
 		if err = util.WaitForFileMod(ctx, fileModified, errChan); err != nil {
 			return err
 		}
+		SetNotReady(isReady)
```

Comment:
do we have logs or something in `WaitForFileMod`? If not I am a bit worried it will be confusing why ready fails

Reply:
There aren't any logs in `WaitForFileMod`. And right it might be better to just leave it as ready until an invalid configuration is detected?

Reply:
I am not as familiar with all of this logic, but if the pod is "NotReady" we need a way to be able to look at the logs and see why. If it just flips to notready its confusing

Reply:
Okay I talked to @stewartbutler about this, and the function now sets the pod to "NotReady" when an invalid configuration (logs invalid config) or error occurs (error will cause the whole process to exit).